### PR TITLE
user validate and calculate results

### PR DESCRIPTION
**Pull request by Bazi99**
issue 62: https://github.com/eduardconstantin/azure-fundamentals/issues/62

### CODE REVIEW

**Comment by Bazi99** [comment]
> @Bazi99 you have some conflicts, you didn't create a new branch from the dev branch

okay i will resolve it today.

**Comment by Bazi99** [comment]
okay i resolved the conflicts

ut 14. 11. 2023 o 18:05 Eduard-Constantin Ibinceanu <
***@***.***> napísal(a):

**Comment by Bazi99** [comment]
I am reminding you if you didnt forget...

st 15. 11. 2023 o 20:13 Daniel Baziľák ***@***.***>
napísal(a):

**Comment by eduardconstantin** [comment]
I get this error when I try to install the dependencies
image

**Comment by Bazi99** [comment]
you have to do "npm install --force", its because testing library for react
hooks, i have to have tests

po 20. 11. 2023 o 17:58 Eduard-Constantin Ibinceanu <
***@***.***> napísal(a):

**Comment by eduardconstantin** [comment]
I see, isn't there any replacement for the testing library? 
I would like to keep only the react-hook-form, I am not familiar with formik. If you tell me what didn't worked I might be able to help.

**Comment by Bazi99** [comment]
I didnt find any replacement for this sorry, With hook form returning back to the previous question didnt load the saved answers and i hadnt clue because i am working with formik always

**Comment by eduardconstantin** [comment]
this is not good, I can't merge the PR with that error and there are a too many changes.
I think you should create separate PRs for functionality and for testing.

**Comment by Bazi99** [comment]
tests are added only in commit with "tests"

**Comment by eduardconstantin** [comment]
I see testing library supports renderHook
[https://testing-library.com/docs/react-testing-library/api#renderhook](https://testing-library.com/docs/react-testing-library/api#renderhook)

**Comment by Bazi99** [comment]
okay i will try this library then, should i create new merge request then?

**Comment by eduardconstantin** [comment]
no, do it here

**Comment by Bazi99** [comment]
done, just removed package and replaced renderHook with you suggestion, working

**Comment by eduardconstantin** [comment]
nicely done, I like what you did with reveal exam feature
image
but you changed too many things in the exam mode UI, there is no skip question button, when I select an answer there is no highlight, the confirmation screen is not correct
You have the figma design files [here](https://www.figma.com/file/ckNeex3dSR7Y32bPfJHiQQ/Azure-Fundamentals?type=design&node-id=417%3A10&mode=design&t=owFkzui3OhqmO7bU-1)

**Comment by Bazi99** [comment]
okay so do i need to do this before accepting merge request?

**Comment by eduardconstantin** [comment]
yes, I can merge it then
The design was already made on the dev branch you can get it from there

**Comment by Bazi99** [comment]
i fixed design

**Comment by eduardconstantin** [comment]
you didn't add the skip question button, remove the previous question and submit buttons
image
image

**Comment by Bazi99** [comment]
I pushed now new commit, hope all things you wanted all fixed... check it out please.

File: .gitignore [hunk 1]
```diff
@@ -12,6 +12,10 @@
 /.next/
 /out/
 
+/.idea
+.idea
+/.idea/
```
Comment (masterHAWK99) [inline .gitignore:17]:
```suggestion
/.idea/
```

File: components/QuizExamForm.tsx [hunk 2]
```diff
@@ -0,0 +1,399 @@
...
+        },
+      ],
+    },
+    onSubmit: () => {
+      saveAnswers(false).then(() => {
+        // eslint-disable-next-line react-hooks/rules-of-hooks
+        const { points } = useResults({
+          questions: questions,
+          answers: savedAnswers,
+        });
```
Comment (masterHAWK99) [inline components/QuizExamForm.tsx:67]: Just move this hook upper, where other hooks are

File: hooks/useResults.tsx [hunk 3]
```diff
@@ -0,0 +1,54 @@
...
+      } else {
+        points = points + (partialPoints > 0 ? (Math.round(partialPoints * 100) / 100) : 0);
+        points = (Math.round(points * 100) / 100);
+      }
+    } else if (noOfAnswers === 1 && !Array.isArray(data.answers[i])) {
+      if (data.questions[i]?.options?.filter((el: any) => el.isAnswer === true)[0]?.text === data.answers[i]) {
+        points = (Math.round((points + 1) * 100) / 100);
+      }
+    }
+  }
```
Comment (masterHAWK99) [inline hooks/useResults.tsx:46]: You have the same code twice, create one function and reuse it

**Review by masterHAWK99 — CHANGES_REQUESTED** [review]
Remove unused and commented code

File: styles/globals.css [hunk 4]
```diff
@@ -1,3 +1,53 @@
...
+    background-position: center;
+    -o-background-size:50% auto;
+    -webkit-background-size:50% auto;
+    -moz-background-size:50% auto;
+    background-size:50% auto;
+}
+
+.points-font {
+    font-size: 48px;
+}
```
Comment (masterHAWK99) [inline styles/globals.css:53]: I would suggest using Tailwind to style instead of using those classes.

**Review by masterHAWK99 — CHANGES_REQUESTED** [review]
Remove unused and commented code

File: styles/globals.css (hunk 4 above)
Comment (eduardconstantin) [reply styles/globals.css:53]: could you refactor this in a new PR?

**Review by eduardconstantin — APPROVED** [review]
Thank you for improving the exam mode!

**Pull request merged**